### PR TITLE
New-app: fix message when language is not detected.

### PR DESCRIPTION
**Pull request by csrwng**
Fixes https://github.com/openshift/origin/issues/2625

### CODE REVIEW

**Comment by csrwng** [comment]
@rhcarvalho ptal

**Comment by smarterclayton** [comment]
LGTM [merge]

File: pkg/generate/app/sourcelookup.go [hunk 1]
```diff
@@ -185,6 +185,10 @@ type SourceRepositoryEnumerator struct {
 	Tester    dockerfile.Tester
 }
 
+// ErrNoLanguageDetected is the error returned when no language can be detected by all
+// source code detectors.
+var ErrNoLanguageDetected = fmt.Errorf("No language matched the source repository")
```
Comment (rhcarvalho) [inline pkg/generate/app/sourcelookup.go]: Seems that we favor `errors.New` in our code base.

**Comment by rhcarvalho** [comment]
Just a comment about using `errors.New` instead of `fmt.Errorf`. LGTM.

**Pull request merged**